src/Symfony/Bundle/TwigBundle/ErrorRenderer/TwigHtmlErrorRenderer.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\ErrorRenderer;
+
+use Symfony\Component\ErrorRenderer\ErrorRenderer\ErrorRendererInterface;
+use Symfony\Component\ErrorRenderer\ErrorRenderer\HtmlErrorRenderer;
+use Symfony\Component\ErrorRenderer\Exception\FlattenException;
+use Twig\Environment;
+use Twig\Error\LoaderError;
+use Twig\Loader\ExistsLoaderInterface;
+
+/**
+ * Provides the ability to render custom Twig-based HTML error pages
+ * in non-debug mode, otherwise falls back to HtmlErrorRenderer.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class TwigHtmlErrorRenderer implements ErrorRendererInterface
+{
+    private $twig;
+    private $htmlErrorRenderer;
+    private $debug;
+
+    public function __construct(Environment $twig, HtmlErrorRenderer $htmlErrorRenderer, bool $debug = false)
+    {
+        $this->twig = $twig;
+        $this->htmlErrorRenderer = $htmlErrorRenderer;
+        $this->debug = $debug;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getFormat(): string
+    {
+        return 'html';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function render(FlattenException $exception): string
+    {
+        if ($exception->getHeaders()['X-Debug'] ?? $this->debug) {
```

Comment:
The current X-Debug header allows enabling debug mode. This is an unessarry security risk as someone might forward the request headers to the exception headers and suddenly people can see the debug output in prod.

Please  change the header so it only allows to disable debug, maybe `Disable-Debug`, as that is what we need in the PreviewErrorController.

Reply:
And please add a test for this.

Reply:
Good capture, although I prefer the current header name for better readability of the code.
Tests added, ensuring that you can't enabling debug mode via X-Debug on non-debug mode.
Thanks!!

Reply:
I've moved the change to https://github.com/symfony/symfony/pull/32693